cmd/storagenode/gracefulexit.go

```diff
@@ -0,0 +1,145 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/spf13/cobra"
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+
+	"storj.io/storj/internal/memory"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/process"
+	"storj.io/storj/pkg/rpc"
+	"storj.io/storj/pkg/storj"
+)
+
+type gracefulExitClient struct {
+	conn *rpc.Conn
+}
+
+func dialGracefulExitClient(ctx context.Context, address string) (*gracefulExitClient, error) {
+	conn, err := rpc.NewDefaultDialer(nil).DialAddressUnencrypted(ctx, address)
+	if err != nil {
+		return nil, errs.Wrap(err)
+	}
+	return &gracefulExitClient{conn: conn}, nil
+}
+
+func (client *gracefulExitClient) getNonExitingSatellites(ctx context.Context) (*pb.GetNonExitingSatellitesResponse, error) {
+	return client.conn.NodeGracefulExitClient().GetNonExitingSatellites(ctx, &pb.GetNonExitingSatellitesRequest{})
+}
+
+func (client *gracefulExitClient) initGracefulExit(ctx context.Context, req *pb.StartExitRequest) (*pb.StartExitResponse, error) {
+	return client.conn.NodeGracefulExitClient().StartExit(ctx, req)
+}
+
+func (client *gracefulExitClient) close() error {
+	return client.conn.Close()
+}
+
+func cmdGracefulExitInit(cmd *cobra.Command, args []string) error {
+	ctx, _ := process.Ctx(cmd)
+
+	ident, err := runCfg.Identity.Load()
+	if err != nil {
+		zap.S().Fatal(err)
+	} else {
+		zap.S().Info("Node ID: ", ident.ID)
+	}
+
+	scanner := bufio.NewScanner(os.Stdin)
+	fmt.Println("Please be aware that by starting a graceful exit on a satellite, you will no longer be allowed to participate in repairs or uploads from that satellite.This action can not be undone. Are you sure you want to continue? y/n")
+	var confirmInput string
+	for scanner.Scan() {
+		confirmInput = scanner.Text()
+		break
+	}
+	confirmationOptions := map[string]struct{}{"y": {}, "yes": {}, "Y": {}, "Yes": {}}
```

Comment:
See how we do a y/n prompt in inspector. Seems like it would be a lot easier than having all this logic: https://github.com/storj/storj/blob/master/cmd/inspector/main.go#L404